src/Panel/ChangesDialog.php

```diff
@@ -31,13 +37,27 @@ public function files(): array
 		return $this->items($this->changes->files());
 	}
 
+	/**
+	 * Helper method to return item props for a single given model
+	 */
+	public function item(File|Page|User $model): array
+	{
+		$item = match (true) {
+			$model instanceof File => new FileItem(file: $model),
+			$model instanceof Page => new PageItem(page: $model),
+			$model instanceof User => new UserItem(user: $model),
+		};
+
+		return $item->props();
+	}
+
 	/**
 	 * Helper method to return item props for the given models
 	 */
 	public function items(Collection $models): array
 	{
 		return $models->values(
-			fn ($model) => $model->panel()->dropdownOption()
+			fn ($model) => $this->item($model)
 		);
```

Comment:
`Nitpick/Suggestion:` I don't know if you would like that, but this could be also written as

```php
return $models->values($this->item(...));
```

I think.

Reply:
Yep, that works.